### PR TITLE
refactor: use alloy's `BaseFeeParams`

### PR DESCRIPTION
**Pull request by rodrigoherrerai**
first draft on [7530](https://github.com/paradigmxyz/reth/issues/7530)

@mattsse just to see if it is going in the right direction.

### CODE REVIEW

File: crates/primitives/src/chain/spec.rs [hunk 1]
```diff
@@ -440,27 +439,19 @@ impl From<Vec<(Hardfork, BaseFeeParams)>> for ForkBaseFeeParams {
...
-    /// Get the base fee parameters for Ethereum mainnet
-    pub const fn ethereum() -> BaseFeeParams {
-        BaseFeeParams {
-            max_change_denominator: EIP1559_DEFAULT_BASE_FEE_MAX_CHANGE_DENOMINATOR,
-            elasticity_multiplier: EIP1559_DEFAULT_ELASTICITY_MULTIPLIER,
-        }
-    }
-
+#[cfg(feature = "optimism")]
+impl OptimismBaseFeeParams for BaseFeeParams {
```
Comment (mattsse) [inline crates/primitives/src/chain/spec.rs:451]: we only use this inside this file, so I'd rather have constants for this:

`const OP_SEPOLIA_BASEFEE_PRARAMs = ...`

**Review by mattsse — CHANGES_REQUESTED** [review]
great!

one suggestion to convert the trait into constants

**Comment by rodrigoherrerai** [comment]
@mattsse modified the trait to constants

**Review by mattsse — CHANGES_REQUESTED** [review]
great!

needs rebase.

I'm unable to make modifications because this pr was opened from main branch, please open future prs from a dedicated branch

**Comment by rodrigoherrerai** [comment]
@mattsse should I open a new pr with a dedicated branch?

**Comment by mattsse** [comment]
that'd would be preferred yes :)

**Pull request closed without merging**